### ISSUE TITLE
spack incorrectly detects slingshot network

### ISSUE THREAD

**Issue by bvanessen** [LLNL/spack#35778]
### Steps to reproduce

In `lib/spack/spack/platforms/cray.py` the function to determine if a slingshot network is present relies on a hardcoded path that exists on systems without slingshot.

```console
def slingshot_network():
    return os.path.exists("/lib64/libcxi.so")
```

### Information on your system

LLNL Pascal

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
The function now looks different:
```python
def slingshot_network():
    return os.path.exists("/opt/cray/pe") and (
        os.path.exists("/lib64/libcxi.so") or os.path.exists("/usr/lib64/libcxi.so")
    )
```
Do you still have this issue?

**Comment by tgamblin** [comment]
I believe this is resolved by #35779.
